component/blackbox-exporter.libsonnet

```
@@ -0,0 +1,125 @@
+// main template for openshift4-slos
+local com = import 'lib/commodore.libjsonnet';
+local kap = import 'lib/kapitan.libjsonnet';
+local kube = import 'lib/kube.libjsonnet';
+local inv = kap.inventory();
+// The hiera parameters for the component
+local params = inv.parameters.openshift4_slos;
+
+local matchLabels = {
+  'app.kubernetes.io/instance': 'prometheus-blackbox-exporter',
+  'app.kubernetes.io/name': params.blackbox_exporter.name,
+};
+
+local labels = matchLabels {
+  'app.kubernetes.io/managed-by': 'commodore',
+  'app.kubernetes.io/component': 'openshift4-slos',
+};
+
+local exporterConfig = std.manifestYamlDoc(params.blackbox_exporter.config, true, false);
+
+local sa =
+  kube.ServiceAccount(params.blackbox_exporter.name) {
+    metadata+: {
+      labels+: labels,
+      namespace: params.blackbox_exporter.namespace,
+    },
+  };
+
+local configMap =
+  kube.ConfigMap(params.blackbox_exporter.name) {
+    metadata+: {
+      labels+: labels,
+      namespace: params.blackbox_exporter.namespace,
+    },
+    data+: {
+      'blackbox.yaml': exporterConfig,
+    },
+  };
+
+local deploy = com.namespaced(
+  params.blackbox_exporter.namespace,
+  kube.Deployment(params.blackbox_exporter.name) {
+    metadata+: {
+      labels+: labels,
+      name: params.blackbox_exporter.name,
+      namespace: params.blackbox_exporter.namespace,
+    },
+    spec+: {
+      replicas: 1,
+      selector: {
+        matchLabels: matchLabels,
+      },
+      template+: {
+        metadata+: {
+          annotations: {
+            'checksum/config': std.md5(exporterConfig),
+          },
+          labels+: labels,
+        },
+        spec: {
```

With `spec+`, you'd get support for using `kube.Pod()` helpers here:

```suggestion
        spec+: {
```